### DISCUSSION TITLE
Rework and decentralize sign up and sign in activity

### DISCUSSION THREAD

**Discussion by qnixsynapse** [element-hq/element-meta#704] [Ideas]
The current Riot implementation is too much centralized, i.e, It chooses matrix.org homeserver by default and selecting any other homeserver requires to specifically type the full address of the homeserver. We can mitigate this by automatically choosing the best homeserver for the user based on user's approximate location/home server health.

Matrix is a decentralized chat platform but the current implementation makes it more centralized. 

# Sign up Screen activity

We can decrease the number of input fields by providing a list of publicly available homeservers and their location. Here is the best way to get a user sign up:
* The screen has four input fields and a list, One is for username and other two is for password(with verification), and the last one is to enter their email address. 
* The list should consist of publicly available homeservers ranked by health and location. The closest/ healthiest server should be automatically selected. Not matrix.org. 
* The location should be determined approximately. If the user do not allow the app to access location, select the healthiest server.

# Sign in Screen activity

This is much simpler. There is a Matrix ui field and password field. User when signing in just needs to enter their matrix user ID. The homeserver is automatically determined by extracting and the domain and re constructing the home server URL. For example,
If the matrix user ID is akarshanbiswas:diasp.in, extract diasp.in and store it in $DOMAIN and add chat as prefix.
==> https://chat.$DOMAIN ==> https://chat.diasp.in (Final construction of home server URL)

For both activity we can also add a button to enter custom Identity server. By default the vector.im server should be automatically selected.

Opinions and suggestions are welcome. 

The goal of this issue is to stop riot messenger from centralizing the matrix platform for new users.

I am mentioning this issue: https://github.com/matrix-org/synapse/issues/2257 since we were discussing in diasp.in team room about the centralization problem in matrix and it came up during the discussion.

This issue can be carry forwarded to riot web/desktop as well.

**Comment by kittykat** [comment]
Moving to discussions as this is cross-platform :+1:
